src/rpc/blockchain.cpp

```diff
@@ -1583,14 +1583,14 @@ UniValue savemempool(const JSONRPCRequest& request)
     if (request.fHelp || request.params.size() != 0) {
         throw std::runtime_error(
             "savemempool\n"
-            "\nDumps the mempool to disk.\n"
+            "\nDumps the mempool to disk. It will fail until the previous dump is fully loaded.\n"
             "\nExamples:\n"
             + HelpExampleCli("savemempool", "")
             + HelpExampleRpc("savemempool", "")
         );
     }
 
-    if (!DumpMempool()) {
+    if (!g_is_mempool_loaded || !DumpMempool()) {
```

Comment:
Would be nice to give a different error message here.

Reply:
Agree, with the corresponding test.

Reply:
Should the `g_is_mempool_loaded` check be inside of `DumpMempool`? No need to see `g_is_mempool_loaded` in the RPC code.

Note that with this change there is no need for a functional test, just a c++ test `DumpMempool`.